addons/mail/static/src/models/user/user.js

```diff
@@ -113,7 +113,9 @@ function factory(dependencies) {
                 thread.model === 'mail.channel' &&
                 thread.public === 'private'
             );
-            if (!chat) {
+            // if chat is already known but not pinned, then it means current user just left it
+            // so it's needed here to repin it and the called RPC do that as a side effect
+            if (!chat ||!chat.isPinned) {
```

Comment:
Would put comment inside the `if`, clearly state that this comment is about `!channel.isPinned`, and explain a bit better why doing a `channel_get` is important (from my understanding, it's to pin chat while also having it marked as pinned server-side).

The current comment is not good. It's not relevant to know which (potential) flows makes this scenario possible.